### ISSUE TITLE
backporting fix to 2.0.8 due for spring 2.x compatibility

### ISSUE THREAD

**Issue by sushilnayak** [jasync-sql/jasync-sql#344]
Hi @oshai , Raising this issue to backport fixes to 2.0.8 so that fixes are available for Spring Boot 2.X applications. 
As we know it will take a lot of time for companies to move to Spring Boot 3.X hence supporting all the fixes( only bugfixes or security enhancements) will be good otherwise it gets little problematic.

Let me know what you think?

**Comment by oshai** [comment]
I think we will backport critical fixes only. But anyway, we will check it on a case by case basis.

Generally speaking, spring pre-3.x uses a non GA version of R2DBC so I prefer not to support it.

**Comment by sushilnayak** [comment]
> Generally speaking, spring pre-3.x uses a non GA version of R2DBC so I prefer not to support it.

I agree, but we can always upgrade spi version (< 1.0.0-release) to support newer jasync dependencies in existing 2.X spring applications, isn't it?. 

My concern also comes from not being able to upgrade jasync beyond 2.0.8 in 2.X spring application. I get a feeling that im missing something if i don't upgrade beyond 2.0.8 and hence the request to support backporting.

**Comment by oshai** [comment]
you can try to use r2dbc 1 with boot 2, but I guess spring team did not test that.

Let's discuss again if there is a concrete issue that needs to be back-ported.

**Comment by SalyczeQ** [comment]
Hello team,
Today I was heading this issue: 
https://github.com/jasync-sql/jasync-sql/issues/300
which seems to be fixed in 2.1.1 so I think this is the first use case which I would really appreciate if it could be backported to 2.0.x?

**Comment by oshai** [comment]
IIUC 2.0.x will work ok with 0.8 so backporting it will break that (unless we have specific branch for 0.9?

**Comment by sushilnayak** [comment]
@oshai can we please backport #348 & #350 (when PR is merged) so that spring 2.x users can benefit from these additions

**Comment by oshai** [comment]
I created a branch r2dbc-pre-ga (from 2.0.8 release).

Since I don't have the time to backport and test the changes to there, I suggest that you'll create PR's for changes you'd like to backport and I'll review, merge and release them on a 2.0.x release stream. Is that sounds good enough?

**Comment by KarboniteKream** [comment]
Would you welcome/accept a contribution to make the 2.0.x series compatible with SPI 0.9? We cannot upgrade to Spring Boot 3 in a reasonable time, and downgrading to 2.6 is also not an option. We can make the PRs and perform any necessary testing.

**Comment by oshai** [comment]
Yes, 0.9 sgtm. If other devs need 0.8 we can have a separate branch for that. We will just need to figure out a versioning schema, but that's feasible.
